### PR TITLE
add status item to show/create packages.yml file

### PR DESCRIPTION
**Pull request by pgrivachev**
- Deleted accidentally duplicated `onDidChangeActiveTextEditor`
- Simplified "Should install latest dbt, restart language server and compile model with new dbt version" test
- Now we pass projectPath to commands
- Added status item to show/create packages.yml file
- Added declarationMap for common project. Now go to definition from other project jump to source file instead of out file
- Fixed multiple parse manifest events in case we got more than one changes in our handler
- Refactored client commands

### CODE REVIEW

File: client/src/commands/OpenOrCreatePackagesYml.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    const column = window.activeTextEditor?.viewColumn;
+    const fileUri = Uri.joinPath(Uri.parse(projectPath), 'packages.yml');
+
+    try {
+      const document = await workspace.openTextDocument(fileUri);
+      await window.showTextDocument(document, column);
+    } catch {
+      const doc = await workspace.openTextDocument(fileUri.with({ scheme: 'untitled' }));
+      await window.showTextDocument(doc, column);
+    }
```
Comment (fivetran-dmitryboykov) [inline client/src/commands/OpenOrCreatePackagesYml.ts:17]: If we can get rid of try/catch here?

File: server/src/StatusSender.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  sendStatus(): void {
+    const statusNotification: StatusNotification = {
+      projectPath: this.projectPath,
+      pythonStatus: {
+        path: this.featureFinder.getPythonPath(),
+      },
+      dbtStatus: {
+        versionInfo: this.featureFinder.versionInfo,
+      },
```
Comment (fivetran-dmitryboykov) [inline server/src/StatusSender.ts:27]: Why we do not send `packagesStatus` ?

File: server/src/FileChangeListener.ts [hunk 3]
```diff
@@ -38,28 +44,38 @@ export class FileChangeListener {
...
   onDidChangeWatchedFiles(params: DidChangeWatchedFilesParams): void {
-    const dbtProjectYmlPath = path.resolve(this.workspaceFolder, DbtRepository.DBT_PROJECT_FILE_NAME);
     const manifestJsonPath = path.resolve(this.workspaceFolder, this.dbtRepository.dbtTargetPath, DbtRepository.DBT_MANIFEST_FILE_NAME);
-    const packagesYmlPath = path.resolve(this.workspaceFolder, DbtRepository.DBT_PACKAGES_FILE_NAME);
     const packagesPaths = this.dbtRepository.packagesInstallPaths.map(p => path.resolve(this.workspaceFolder, p));
 
+    // For some paths we want to do action only once even we got several changes here
+    if (this.containsChangeWithPath(params.changes, manifestJsonPath)) {
+      this.updateManifestNodes();
+    }
```
Comment (fivetran-dmitryboykov) [inline server/src/FileChangeListener.ts:53]: If we can use the same logic for `packagesPaths` ?

**Review by fivetran-dmitryboykov — COMMENTED** [review]
Looks good.

File: client/src/commands/OpenOrCreatePackagesYml.ts (hunk 1 above)
Comment (pgrivachev) [reply client/src/commands/OpenOrCreatePackagesYml.ts:17]: If we can't open document `openTextDocument` throws an exception. And in this case we create document ourself.

File: server/src/StatusSender.ts (hunk 2 above)
Comment (pgrivachev) [reply server/src/StatusSender.ts:28]: We send it separately from `sendPackagesStatus` method. Not sure that we need duplicate.

File: server/src/FileChangeListener.ts (hunk 3 above)
Comment (pgrivachev) [reply server/src/FileChangeListener.ts:53]: We can. I did it. And not sure now, do we need debounce logic?

**Pull request merged**